### ISSUE TITLE
awxweb:8052 not found when running job.

### ISSUE THREAD

**Issue by TemperingPick** [ansible/awx#135]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

 - API

##### SUMMARY

404 Error when running a job against an inventory, hostname not found.

##### ENVIRONMENT

* AWX version: devel
* Ansible version:  2.3.2
* Operating System: RHEL7
* Web Browser: Chrome 60.x

##### STEPS TO REPRODUCE

Run job with known up host.

##### EXPECTED RESULTS

Job should run.

##### ACTUAL RESULTS

ERROR! Attempted to execute "/usr/lib/python2.7/site-packages/awx/plugins/inventory/awxrest.py" as inventory script: Inventory script (/usr/lib/python2.7/site-packages/awx/plugins/inventory/awxrest.py) had an execution error: 404 Client Error: Not Found for url: http://awxweb:8052/api/v1/inventories/2/script/?hostvars=1

HTTP 404: Your requested host "awxweb" could not be resolved by DNS. The document at the specified URL does not exist.

##### ADDITIONAL INFORMATION

[2017-09-12 15:32:30,083: ERROR/MainProcess] Task awx.main.tasks.run_job[6a555e88-0891-4446-84e6-fb82d4a7b70e] raised unexpected: TaskError('%s encountered an error (rc=%s), please see task stdout for details.',)
Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib/python2.7/site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/var/lib/awx/venv/awx/lib/python2.7/site-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/awx/main/tasks.py", line 466, in _wrapped
    return f(self, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/awx/main/tasks.py", line 905, in run
    raise TaskError(instance, rc)
TaskError: %s encountered an error (rc=%s), please see task stdout for details.

**Comment by wenottingham** [comment]
Are you running in local docker, minishift, openshift, or....?

**Comment by TemperingPick** [comment]
Local docker.

**Comment by theasp** [comment]
There appears to be a hard requirement that the web container have the hostname `awxweb` and the task container `awx`.  If you built yours using different container names, you will have to adjust your links so that `awxweb` on your task container resolves to your web container, even if you have them on the same docker network.  Note that hostnames don't get added to `/etc/hosts` without links.

**Comment by TemperingPick** [comment]
I built mine from the playbook and it called the images awx_web and awx_task. I didn't rename them.

**Comment by theasp** [comment]
Right, but it wants `awxweb` to resolve, not `awx_web`.  I'm not saying the playbook is right or wrong, but your error message would indicate that it can't resolve that name.

**Comment by TemperingPick** [comment]
What all do I need to rename to fix the issue? I've renamed the docker containers to awx from awx_tasks and awxweb from awx_web.

**Comment by matburt** [comment]
@TemperingPick what does the `awx_web` container think its hostname is?

**Comment by matburt** [comment]
I totally could see that this *could* be a problem... and I've heard it a few times now, I'm just trying to track down how we could find ourselves int his scenario.

**Comment by TemperingPick** [comment]
@matburt right now it says awxweb after I changed it.

**Comment by leifmadsen** [comment]
This is related to #99 and #53 where I state that the file of hard coded values should be migrated to a template that allows you to specify things like a hostname that you expect.

The work around is to place `awxweb` into your `/etc/hosts` file pointing at the IP address of your server hosting AWX.

**Comment by leifmadsen** [comment]
@TemperingPick see #53 and my note above, but the name of the container isn't an issue at all here. The `/etc/tower/settings.py` file has several hard coded arguments, including the hostname, so you'll need to adjust your `/etc/hosts` file to point at the server IP address, and access the AWX service via `http://awxweb` in your browser.

**Comment by leifmadsen** [comment]
Specifically here:  https://github.com/ansible/awx/blob/devel/installer/image_build/files/settings.py#L24

I had to modify the value as well to remove the port address, since by default port 80 on the `awx_web` already forward to the `awx_task` container over the docker network (so you don't really have direct access to port 5062).

**Comment by theasp** [comment]
The hostname is only visible inside the same container, but container names are resolvable on their own.    You can have an entry added to `/etc/hosts` for you by setting up a link with a hostname. 

@leifmadsen solved the problem by changing `settings.py`, I solved the problem by providing the names that settings.py is expecting.  Both should work, and it goes without saying that the hard coded name is a problem. :)

For example, I am doing something like this:
```yaml
- docker_container:
    name: awx_web
    hostname: awxweb

- docker_container:
    name: awx_task
    hostname: awx
    links:
      - "awx_web:awxweb"  # An entry in /etc/hosts will be added so that awxweb has the address for awx_web
```

**Comment by matburt** [comment]
I definitely prefer your solution here @theasp ... if you have a chance then submit a PR for a generalized case.

**Comment by matburt** [comment]
@TemperingPick the hostname is not the same thing as the name of the container...

you can see here https://github.com/ansible/awx/blob/devel/installer/local_docker/tasks/main.yml#L174

When we launch the container we tell it to have that hostname and then we provide the link to the awx_task container and (hopefully? most of the time...) it can resolve the web container from the task container.

That appears to not be happening, so I'm wondering what does the web container think its hostname is (by execing to the container directly and seeing) and why, in some circumstances, can't the task container see it as that name that we gave.

**Comment by TemperingPick** [comment]
@matburt Sorry if I didn't make it clear enough but I did exec into the container and the hostname was awxweb for the awx_web container, and awx for the awx_task container.

**Comment by matburt** [comment]
Well that's good at least! Now to figure out why the task container can't reference it under some circumstances.

**Comment by TemperingPick** [comment]
I can ping awx from awx_web and vice versa while in the container. Now when I run a job, it does give me my company's 404 page so is it possible it's trying to route outside of the container?

**Comment by theasp** [comment]
@matburt I'm not sure my solution is a good idea!  It would be much cleaner to just get the name of the other container from an environment variable.  No need to use links or worry about having the right name.

**Comment by matburt** [comment]
I may try just adding that link in that you have
```
- "awx_web:awxweb"  # An entry in /etc/hosts will be added so that awxweb has the address for awx_web
```

**Comment by matburt** [comment]
hopefully fixed in #162
